### PR TITLE
Network

### PR DESCRIPTION
**Pull request by Vivelis**

### CODE REVIEW

File: src/client/Client.cpp [hunk 1]
```diff
@@ -11,9 +11,11 @@
 #include "GameEngine.hpp"
 #include "../network_c/Network.hpp"
 
+#include "../ecs/ComponentsArray/Components/Components.hpp"
+#include "../ecs/ComponentsArray/Systems/System.hpp"
+
```
Comment (Vivelis) [inline src/client/Client.cpp:16]: To remove (not needed)

File: src/ecs/ComponentsArray/Components/Components.hpp [hunk 2]
```diff
@@ -86,6 +86,27 @@ struct Score {
     int score;
 };
 
+struct ComponentIN {
+    //void *ptrToStructToReceive;
```
Comment (Vivelis) [inline src/ecs/ComponentsArray/Components/Components.hpp:90]: To remove (no comment to hide code)

File: src/ecs/ComponentsArray/Components/Components.hpp [hunk 3]
```diff
@@ -86,6 +86,27 @@ struct Score {
...
+    struct Health;
+    struct Damage;
+    struct Score;
+    struct Text;
+    struct Drawable;
+    struct Control;
+    struct Pattern;
+    struct Hitbox;
+};
+
```
Comment (Vivelis) [inline src/ecs/ComponentsArray/Components/Components.hpp:109]: Should be moved to network folder

File: src/ecs/ComponentsArray/Components/Components.hpp [hunk 4]
```diff
@@ -118,4 +139,25 @@ struct Hitbox {
...
+    Health health;
+    Damage damage;
+    Score score;
+    Text text;
+    Drawable drawable;
+    Control control;
+    Pattern pattern;
+    Hitbox hitbox;
+};
+
```
Comment (Vivelis) [inline src/ecs/ComponentsArray/Components/Components.hpp:162]: Should be moved to network folder

File: src/ecs/ComponentsArray/Systems/System.cpp [hunk 5]
```diff
@@ -54,6 +54,11 @@ void System::draw_system(registry &r, sf::RenderWindow &window)
     }
 }
 
+void System::setNetwork(Network *ptrNetwork)
+{
+    _ptrNetwork = ptrNetwork;
+}
+
```
Comment (Vivelis) [inline src/ecs/ComponentsArray/Systems/System.cpp:61]: Should be moved to network folder

File: src/ecs/ComponentsArray/Systems/System.hpp [hunk 6]
```diff
@@ -22,6 +23,8 @@ class System {
         void control_system(registry &r);
         void velocity_system(registry &r, sf::Time &elapsed);
         void hitbox_system(registry &r);
+        void send_system(registry &r);
+        void setNetwork(Network *ptrNetwork);
```
Comment (Vivelis) [inline src/ecs/ComponentsArray/Systems/System.hpp:27]: No network in game systems

File: src/ecs/ComponentsArray/Systems/System.hpp [hunk 7]
```diff
@@ -35,6 +38,8 @@ class System {
     protected:
     private:
         std::unordered_map<std::string, sf::Texture> _textures;
+        Network *_ptrNetwork = NULL;
+
```
Comment (Vivelis) [inline src/ecs/ComponentsArray/Systems/System.hpp:42]: No network in game system

File: src/network_c/Network.cpp [hunk 8]
```diff
@@ -6,17 +6,24 @@
 */
 #include "Network.hpp"
 #include "TestClassPlayer.hpp"
```
Comment (Vivelis) [inline src/network_c/Network.cpp:8]: Don't exist anymore

File: src/network_c/TestStructPlayer.hpp [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    const char *name;
+    std::string level;
+    int hp;
+    int armor;
+    bool drip;
+    char c;
+    char array[30];
+};
+
+#endif /* !TESTSTRUCTPLAYER_HPP_ */
```
Comment (Vivelis) [inline src/network_c/TestStructPlayer.hpp:23]: no test struct

File: src/network_s/TestStructPlayer.hpp [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+struct testPlayer2 {
+    const char *name;
+    std::string level;
+    int hp;
+    int armor;
+    bool drip;
+    char c;
+    char array[30];
+};
+
```
Comment (Vivelis) [inline src/network_s/TestStructPlayer.hpp:22]: no test struct

**Review by floazema — APPROVED** [review]

**Pull request merged**